### ISSUE TITLE
Removing global to make it gradio usable across multiple sessions

### ISSUE THREAD

**Issue by xhluca** [ServiceNow/AgentLab#186]
https://github.com/ServiceNow/AgentLab/blob/3c123811238879ff117c0d5b95d1da2db2101524/src/agentlab/analyze/agent_xray.py#L591C5-L591C16

By using global, we cannot use agent xray across multiple tabs

**Comment by recursix** [comment]
using module variables is generally not a good practice. With gradio, it's the only way I found to efficiently pass an object to the functions without it being a gradio data. Removing global would break the whole UI. 

To achieve multi-tabs, you need to launch multiple gradio instance. I think we would have to let it chose a port though ([here](https://github.com/ServiceNow/AgentLab/blob/3c123811238879ff117c0d5b95d1da2db2101524/src/agentlab/analyze/agent_xray.py#L504))

**Comment by recursix** [comment]
perhaps something like this:
```python
import os

do_share = os.getenv("AGENTXRAY_SHARE_GRADIO", "false").lower() == "true"
port_env = os.getenv("AGENTXRAY_APP_PORT", None)

if port_env:
    # Use the specified port
    demo.launch(server_port=int(port_env), share=do_share)
else:
    # Let Gradio pick the port
    demo.launch(share=do_share)
```

**Comment by xhluca** [comment]
Thanks, I am currently launching multiple instances, which mitigates the problem.

One concern is that the xray ui might be used by multiple viewers (e.g. huggingface spaces), it might not be possible to spin 10 instances if there are 10 viewers, for example.

> without it being a gradio data

Thanks, is there any concern why we should avoid using gradio data? I have found [gradio state](https://www.gradio.app/docs/gradio/state) to work fairly well!

**Comment by recursix** [comment]
I made tests in the past with radio state and it was clunky. Which is why I decided to go with a global. I don't remember what was the issue with radio state though.

**Comment by xhluca** [comment]
perhaps they fixed it in the newest gradio. however changing from global to state would take a lot of refactoring effort. understandable that we don't want to do it right now (perhaps in the future when sharing an xray with a lot of users on spaces)

**Comment by ThibaultLSDC** [comment]
Now that we have such spaces for showcase ([gradio demo](https://huggingface.co/spaces/agentlabtraces/agentxray)), maybe we could restart a discussion on the topic ?

**Comment by xhluca** [comment]
Agree. I'm sharing agent X-ray results to collaborators but have to tell them to refresh if they run into issues with desync. It would be a great ux improvement to fix global
